app/javascript/components/KeywordSearch.js

```diff
@@ -1,62 +1,39 @@
-import React, { Component } from 'react';
+import React from 'react';
 import Button from 'react-bootstrap/lib/Button';
 import InputGroup from 'react-bootstrap/lib/InputGroup';
 import Form from 'react-bootstrap/lib/Form';
 import { faSearch } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 
 
-class KeyWordSearch extends React.Component{
-  constructor(props){
-    super(props);
-    this.handleSubmit = this.handleSubmit.bind(this)
-    this.state = {
-      searchTerms: undefined,
-    };
-  };
+export default function KeywordSearch (props){
 
-  handleSubmit(e){
+  const handleSubmit = (event) => {
     // Prevent full page reload
-    e.preventDefault();
-
-    const searchTerm = e.target.elements.searchText.value.trim();
+    event.preventDefault();
+    const searchTerm = event.target.elements.keywordText.value.trim();
     if(searchTerm){
-      // Need to check if search terms are empty and if prevstate is different
-      // From current state
-      this.setState(()=>{
-        return {
-          searchTerms:searchTerm
-        };
-
-      });
-      this.setState((prevState)=>{
-        return {
-          searchTerms:prevState.searchTerms
-        };
-      });
-    }
+      props.updateKeyword(searchTerm);
+    };
   }
-
-  render(){
-    return(
-      <div>
-      <Form onSubmit = {this.handleSubmit}>
-        <InputGroup>
-          <input 
-          type="text" 
-          placeholder="Enter keyword" 
-          name="searchText"/>
-            <Button className="search-button" onClick={this.handleSubmit}>
+  
+  return (
+    <div id='keyword-search'>
+      <Form horizontal onSubmit = {handleSubmit} >
+        <InputGroup id='keyword-input-group'>
+          <input
+            id="keyword-input"
+            type="text" 
+            placeholder="Enter Keyword" 
```

Comment:
i think this is more readable as "Enter keyword"

Reply:
055e532